### PR TITLE
#597 Services sidebar - hide unnecessary headings

### PR DESCRIPTION
**Pull request by jankaBehal**
Remove all sections from table of content by default
Use data-attributes (='toc' for whole section, ='toc-heading' for heading only)
Add following sections to be displayed in Table of content
- FAQ (heading only)
- CardsList (Odkazy) (heading only)
- Markdown (whole section)

Snímka obrazovky (45)

### CODE REVIEW

File: next/src/components/common/TableOfContents/useHeadings.ts [hunk 1]
```diff
@@ -2,8 +2,30 @@ import { useIsClient } from 'usehooks-ts'
 
 type HeadingLevels = 'h2' | 'h3' | 'h4' | 'h5' | 'h6'
 
+// used in section wrapper that needs to be used for heading in table of content
+export const ATTRIBUTE_TABLE_OF_CONTENT = 'toc'
+
+// used in all section headers for heading in table of content
+export const ATTRIBUTE_TABLE_OF_CONTENT_HEADING = 'toc-heading'
```
[the remark this answers is not in the document]
Comment (jankaBehal) [reply next/src/components/common/TableOfContents/useHeadings.ts]: fixed

File: next/src/components/sections/FaqSection.tsx [hunk 2]
```diff
@@ -38,7 +36,12 @@ const FaqSection = ({ section }: Props) => {
     // TODO padding-y should probably be managed by the SectionContainer
     <SectionContainer background={backgroundColorFaq} className="py-6 lg:py-18">
       <div className="flex flex-col gap-6">
-        <SectionHeader title={title} text={text} showMoreLink={showMoreLink} />
+        <SectionHeader
+          dataAttribute="toc-heading"
```
[the remark this answers is not in the document]
Comment (jankaBehal) [reply next/src/components/sections/FaqSection.tsx]: Yes I agree, I made some renaming changes

File: next/src/components/layout/Section/SectionHeader.tsx [hunk 3]
```diff
@@ -16,6 +17,7 @@ type SectionHeaderProps = {
   showMoreLink?: LinkFragment | null | undefined
   className?: string
   additionalComponent?: ReactNode
+  dataTableOfContent?: string
```
Comment (radoslavzeman) [inline next/src/components/layout/Section/SectionHeader.tsx:20]: Is it still used or can be removed?

**Review by radoslavzeman — APPROVED** [review]
💯

**Pull request merged**